taskoutput/evergreen_sender.go

```diff
@@ -0,0 +1,232 @@
+	// Local is the sender for "fallback" operations and to collect any
+	// logger error output.
+	Local send.Sender
+	// MaxBufferSize is the maximum number of bytes to buffer before
+	// persisting log data. Defaults to 10MB.
+	MaxBufferSize int
+	// FlushInterval is time interval at which to flush log lines,
+	// regardless of whether the max buffer size has been reached. A flush
+	// interval equal to 0 will disable timed flushes.
+	FlushInterval time.Duration
+
+	// the injectable line parser allows the sender to be agnostic to the
+	// raw log line formats it ingests. Defaults to a basic line parser
+	// that adds the raw string as the log line data field.
+	parse       logLineParser
+	appendLines logLineAppender
+}
+
+func (opts *EvergreenSenderOptions) validate() error {
+	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(opts.appendLines == nil, "must provide an appender function")
+	catcher.NewWhen(opts.MaxBufferSize < 0, "max buffer size cannot be negative")
+	catcher.NewWhen(opts.FlushInterval < 0, "flush interval cannot be negative")
+
+	if opts.parse == nil {
+		opts.parse = func(rawLine string) (log.LogLine, error) {
+			return log.LogLine{Data: rawLine}, nil
+		}
+	}
+
+	if opts.Local == nil {
+		opts.Local = send.MakeNative()
+		opts.Local.SetName("local")
+	}
+
+	if opts.MaxBufferSize == 0 {
+		opts.MaxBufferSize = defaultMaxBufferSize
+	}
+
+	return catcher.Resolve()
+}
+
+// evergreenSender implements the send.Sender interface for persisting
+// Evergreen logs.
+type evergreenSender struct {
+	mu         sync.Mutex
+	ctx        context.Context
+	cancel     context.CancelFunc
+	opts       EvergreenSenderOptions
+	buffer     []log.LogLine
+	bufferSize int
+	lastFlush  time.Time
+	closed     bool
+	*send.Base
+}
+
+// newEvergreeSender creates a new sender for Evergreen logs.
+func newEvergreenSender(ctx context.Context, name string, opts EvergreenSenderOptions) (*evergreenSender, error) {
+	if err := opts.validate(); err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	s := &evergreenSender{
+		ctx:    ctx,
+		cancel: cancel,
+		opts:   opts,
+		Base:   send.NewBase(name),
+	}
+
+	if err := s.SetErrorHandler(send.ErrorHandlerFromSender(s.opts.Local)); err != nil {
+		return nil, errors.Wrap(err, "setting default error handler")
+	}
+
+	if opts.FlushInterval > 0 {
+		go s.timedFlush()
+	}
+
+	return s, nil
+}
+
+// Send sends the given message to the backing log service. This function
+// buffers the messages until the maximum allowed buffer size is reached, at
+// which point the messages in the buffer are written to persistent storage by
+// the backing log service. Send is thread safe.
+func (s *evergreenSender) Send(m message.Composer) {
+	ts := time.Now().UnixNano()
+
+	if !s.Level().ShouldLog(m) {
+		return
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.closed {
+		s.opts.Local.Send(message.NewErrorMessage(level.Error, errors.New("cannot call Send on a closed sender")))
```

Comment:
You did! This was my response:
>I don't like to cause it forces us to use an error and a message.Composer which feels like overkill here since we can just log these errors.